soc/nxp/mcx/mcxnx4x/soc.c

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief System/hardware module for nxp_mcxn94x platform
+ *
+ * This module provides routines to initialize and support board-level
+ * hardware for the nxp_mcxn94x platform.
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/init.h>
+#include <soc.h>
+
+#ifdef CONFIG_PLATFORM_SPECIFIC_INIT
+
+/* This function is taken from the SDK */
+static void system_init(void)
```

Comment:
Can we call `SystemInit` directly here? We also may need to handle stack initialization here if we call any functions